### PR TITLE
Made JavaScriptCatalogHonor honor plural number.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by wkornewald** [comment]
If I'm not mistaken this would be more consistent with Python's gettext behavior, so looks like a good idea.

Minor suggestion: Personally, I only use `@property` when caching the result or when the computation is very efficient. Getter methods would be slightly more straightforward in this case.

**Comment by claudep** [comment]
Would you just removing the property decorator or does prefixing `get_` à la Java adding any value?

**Comment by wkornewald** [comment]
The rest of Django seems to use `get_` prefixes, so I'd say this would be more consistent. However, I'm not a core developer, so maybe someone else can provide better advice. :)

**Comment by claudep** [comment]
@timgraham @charettes , any design opinion on this?

File: django/views/i18n.py [hunk 1]
```diff
@@ -234,12 +235,31 @@ def get_paths(self, packages):
         # paths of requested packages
         return [os.path.join(app.path, 'locale') for app in app_configs]
 
-    def get_plural(self):
-        plural = None
+    @property
+    def num_plurals(self):
+        """
+        Return the number of plurals for this catalog language.
```
Comment (timgraham) [inline django/views/i18n.py]: Instead of two sentences for these docstrings, I'd say: "Return ...., or 2 if ..."

**Comment by charettes** [comment]
I feel like we've always favored `@property` when the only required argument was `self`, `get_` feels Java-esque to me as well.

The only occurrences of `get_` for _getters_ that don't accept a any arguments I can think of from the top of my head are ones in `View` and `BaseModelAdmin` subclasses but they were designed this way to allow defining a static value (e.g. `get_queryset` and `queryset`).

**Comment by claudep** [comment]
Thanks for the inputs, I think I will let the property decorator. I marked the two properties private with `_`-prefixing.

File: django/views/i18n.py [hunk 2]
```diff
@@ -234,12 +235,31 @@ def get_paths(self, packages):
...
+        match = re.search(r'nplurals=\s*(\d+)', self._plural_string or '')
+        if match:
+            return int(match.groups()[0])
+        return 2
+
+    @property
+    def _plural_string(self):
+        """
+        Return the plural string (including nplurals) for this catalog language.
+        Return None if no plural string is available.
```
Comment (timgraham) [inline django/views/i18n.py]: , or None...

**Pull request closed without merging**